### PR TITLE
Add cmake-no-system recipe

### PR DESCRIPTION
**Pull request by jjhelmus**
Add cmake-no-system recipe.  This is a build of cmake built without system libraries, using the versions of these packages included with the cmake source itself.
This is needed to build packages with a cmake build system which are
dependencies of cmake, for example libssh2.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by jjhelmus** [comment]
Do any of the @conda-forge/curl-feedstock maintainers want to be maintainers on this recipe as well.

**Comment by jjhelmus** [comment]
For reference this is the similar to the `curl-feedstock` recipe with `bld.bat` removed since there is no need for a windows version of the package and the following diff:

``` diff
--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -6,6 +6,7 @@ set -ex
              --prefix="${PREFIX}" \
              --system-libs \
              --no-qt-gui \
+             --no-system-curl \
              --no-system-libarchive \
              --no-system-jsoncpp \
              --parallel=${CPU_COUNT} \
diff --git a/recipe/meta.yaml b/recipe/meta.yaml
index cd707aa..93b12db 100644
--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -1,7 +1,7 @@
 {% set version = "3.16.4" %}

 package:
-  name: cmake
+  name: cmake-nocurl
   version: {{ version }}

 source:
@@ -22,9 +22,7 @@ source:

 build:
   number: 0
-  skip: True  # [win and vc<14]
-  ignore_run_exports:
-    - vc
+  skip: True  # [win]

 requirements:
   build:
@@ -37,17 +35,16 @@ requirements:

   host:
     - bzip2           # [unix]
-    - libcurl         # [unix]
     - expat           # [unix]
     - ncurses         # [unix]
     - xz              # [unix]
     - zlib            # [unix]
     - libuv           # [unix]
     - rhash           # [unix]
+    - openssl         # [unix]

   run:
     - bzip2           # [unix]
-    - libcurl         # [unix]
     - expat           # [unix]
     - ncurses         # [unix]
     - xz              # [unix]
@@ -81,7 +78,7 @@ about:
     - Utilities/cmlibrhash/COPYING      # [win]
     - Utilities/cmlibarchive/COPYING
     - Utilities/cmjsoncpp/LICENSE
-  summary: CMake is an extensible, open-source system that manages the build process
+  summary: A CMake built without cURL for use when building cURL's dependencies like libssh2.

 extra:
   recipe-maintainers:
```

**Comment by jjhelmus** [comment]
Looks to be failing because of an issue with conda-build.

**Comment by jakirkham** [comment]
Yeah we just ran into that. Please see PR ( https://github.com/conda/conda-build/pull/3912 ) and xrefs for context.

**Comment by isuruf** [comment]
How about using `cmake-binary` package for dependencies of `cmake`?

**Comment by jjhelmus** [comment]
> How about using `cmake-binary` package for dependencies of `cmake`?

`cmake-binary` is not available in conda-forge and the version in [`defaults`](https://github.com/AnacondaRecipes/aggregate/tree/master/cmake-binary-feedstock/recipe) repackages existing conda packages.  It is also not available for `linux-aarch64`

From this discussion in conda-forge/libssh2-feedstock#35 a cmake package built without cURL seemed like a better approach.

**Comment by isuruf** [comment]
Ok. How about making this package depend on no system packages (except for libgcc-ng, libstdcxx-ng) and call it `cmake-binary` or `cmake-no-system`?

**Comment by jjhelmus** [comment]
Sounds like a good idea @isuruf. I went with `cmake-no-system` to avoid confusion with the existing `cmake-binary` packages.

**Comment by jjhelmus** [comment]
> * Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)

This is vendoring a number of packages (bzip2, curl, expat, etc).  This is intentional. The licenses for vendored packages are included.

**Comment by beckermr** [comment]
@jjhelmus @isuruf where are we on this?

File: recipes/cmake-nocurl/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+test:
+  commands:
+    - cmake --version
+    - ctest --version
+    - cpack --version
+
+about:
+  home: http://www.cmake.org/
+  license: BSD 3-clause
```
Comment (beckermr) [inline recipes/cmake-nocurl/meta.yaml:34]:
```suggestion
  license: BSD-3-Clause
```

File: recipes/cmake-nocurl/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    - Utilities/cmliblzma/COPYING
+    - Utilities/cmlibrhash/COPYING
+    - Utilities/cmlibuv/LICENSE
+    - Utilities/cmzlib/Copyright.txt
+    - Utilities/cmzstd/LICENSE
+  summary: CMake built without system libraries for use when building CMake dependencies.
+
+extra:
+  recipe-maintainers:
+    - jjhelmus
```
Comment (beckermr) [inline recipes/cmake-nocurl/meta.yaml:52]:
```suggestion
    - jjhelmus
    - beckermr
```

count me in to help out if I can!

**Comment by beckermr** [comment]
For other ppl reading this, this is blocked on a conda-build release (likely 3.19.3).

**Comment by wolfv** [comment]
Any news on this? I am looking into libssh2 as well.

**Comment by xhochy** [comment]
Merged https://github.com/conda-forge/conda-build-feedstock/pull/136, so we should retrigger this in ~1h

**Pull request merged**